### ISSUE TITLE
Safe RL using PID: langrangian ppo objective issue

### ISSUE THREAD

**Issue by akjayant** [astooke/rlpyt#201]
In https://github.com/astooke/rlpyt/blob/master/rlpyt/projects/safe/cppo_pid.py - 
if self.objective_penalized:
            # This, or just add c_advantage into advantage?
            c_surr_1 = ratio * c_advantage
            c_surr_2 = clipped_ratio * c_advantage
            c_surrogate = torch.max(c_surr_1, c_surr_2)
            c_loss = self.cost_penalty * valid_mean(c_surrogate, valid)
            if self.use_beta_kl:
                c_loss *= self._beta_kl
            elif self.use_beta_grad:
                c_loss *= self._beta_grad
            if self.diff_norm:  # (1 - lam) * R + lam * C
                pi_loss *= (1 - self.cost_penalty)
                pi_loss += c_loss
            elif self.sum_norm:  # 1 / (1 + lam) * (R + lam * C)
                pi_loss += c_loss
                pi_loss /= (1 + self.cost_penalty)
            else:
                pi_loss += c_loss

shoudn't this be pi_loss -= c_loss since self.cost_penalty is also not negative previously.

**Comment by akjayant** [comment]
Okay just saw it -r_loss
